### PR TITLE
update(code): rabc all function code is added on sdk

### PR DESCRIPTION
**Pull request by rohanranjan250613**

### CODE REVIEW

File: packages/sdk/src/modules/base/index.ts [hunk 1]
```diff
@@ -63,67 +72,73 @@ export class Base {
...
+            !subscribed
+          ) {
+            subscribed = true;
+            console.log('====yahoo====');
+            let inclusionBlockHash;
+            if (result.status.isInBlock) {
+              inclusionBlockHash = result.status.asInBlock.toString();
+            } else if (result.status.isFinalized) {
+              inclusionBlockHash = result.status.asFinalized.toString();
+            }
```
Comment (irediaes) [inline packages/sdk/src/modules/base/index.ts]: Instead of using `else if` just use `if`

File: packages/sdk/src/modules/base/index.ts [hunk 2]
```diff
@@ -134,37 +149,195 @@ export class Base {
...
+            ) {
+              // Handle transaction inclusion
+              subscribed = true;
+
+              let inclusionBlockHash;
+              if (result.status.isInBlock) {
+                inclusionBlockHash = result.status.asInBlock.toString();
+              } else if (result.status.isFinalized) {
+                inclusionBlockHash = result.status.asFinalized.toString();
+              }
```
Comment (irediaes) [inline packages/sdk/src/modules/base/index.ts]: Instead of using `else if` just use `if`

File: packages/sdk/src/modules/base/index.ts [hunk 3]
```diff
@@ -134,37 +149,195 @@ export class Base {
...
+                  }
+
+                  if (lastBlockNumber.gte(executionBlockNr)) {
+                    const blockHash = await api.rpc.chain.getBlockHash(
+                      executionBlockNr
+                    );
+                    const blockHeader = await api.rpc.chain.getHeader(
+                      blockHash
+                    );
+                    const extinsics: GenericExtrinsic<AnyTuple>[] = (
```
Comment (irediaes) [inline packages/sdk/src/modules/base/index.ts:194]: typo error; "extinsics" is missing an "r"

File: packages/sdk/src/modules/base/index.ts [hunk 4]
```diff
@@ -134,37 +149,195 @@ export class Base {
...
+                  if (executionBlockNr.gt(executionBlockStopNr)) {
+                    // Transaction not executed within expected blocks
+                    unsubscribeNewHeads();
+                    reject(
+                      `Tx([${extrinsics.hash.toString()}]) was not executed in blocks: ${executionBlockStartNr.toString()}..${executionBlockStopNr.toString()}`
+                    );
+                    unsub();
+                    return;
+                  }
+
```
Comment (irediaes) [inline packages/sdk/src/modules/base/index.ts:186]: Please a comment block here to explain your code.

File: packages/sdk/src/modules/base/index.ts [hunk 5]
```diff
@@ -134,37 +149,195 @@ export class Base {
...
+                    //       name: ""
+                    //     }
+                    //   } as PeaqEvent;
+                    // });
+                    resolve(eventsTriggeredByTx);
+                    unsub();
+                  }
+                }
+              );
+            } else if (result.isError) {
```
Comment (irediaes) [inline packages/sdk/src/modules/base/index.ts]: Instead of using `else if` just use `if`

File: packages/sdk/src/modules/rbac/rbac.spec.ts [hunk 6]
```diff
@@ -20,52 +21,76 @@ describe('RBAC', () => {
...
-    });
+    //   it('should fetch roles', async () => {
+    //     const response = await rbac.fetchRoles(alice.address);
+    //     expect(typeof response).toBe('object');
+    //     if (response.length > 0) {
+    //       expect(typeof response[0].id).toBe('string');
+    //       expect(typeof response[0].name).toBe('string');
+    //       expect(typeof response[0].enabled).toBe('boolean');
+    //     }
+    //   });
```
Comment (irediaes) [inline packages/sdk/src/modules/rbac/rbac.spec.ts]: Why are you commenting the tests?

File: packages/sdk/src/modules/rbac/index.ts [hunk 7]
```diff
@@ -92,4 +518,783 @@ export class RBAC extends Base {
...
+    | undefined
+  > {
+    try {
+      const { userId, groupId, address = '', seed = '' } = option;
+      if (!groupId) throw new Error('Group Id is required');
+      if (!userId) throw new Error('User Id is required');
+      if (groupId && groupId.length !== 32)
+        throw new Error('Group Id should be 32 char only');
+      if (userId && userId.length !== 32)
+        throw new Error('User Id should be 32 char only');
```
Comment (irediaes) [inline packages/sdk/src/modules/rbac/index.ts]: Too many repetition of these lines of code in different functions.

Create a validation function and call it instead.

**Review by irediaes — CHANGES_REQUESTED** [review]
Seems not to write tests for all implementation. Please write tests.

Also check comments below

File: packages/sdk/src/modules/base/index.ts (hunk 1 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/base/index.ts]: done

File: packages/sdk/src/modules/base/index.ts (hunk 2 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/base/index.ts]: done

File: packages/sdk/src/modules/base/index.ts (hunk 4 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/base/index.ts:186]: done

File: packages/sdk/src/modules/base/index.ts (hunk 5 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/base/index.ts]: done

File: packages/sdk/src/modules/base/index.ts (hunk 3 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/base/index.ts:194]: done

File: packages/sdk/src/modules/rbac/rbac.spec.ts (hunk 6 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/rbac/rbac.spec.ts]: changes are done

File: packages/sdk/src/modules/rbac/index.ts (hunk 7 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/rbac/index.ts]: make a validation function

**Review by kamrangul2404 — APPROVED** [review]

**Pull request merged**